src/middlewares/isWorkspaceMember.middleware.js

```diff
@@ -0,0 +1,33 @@
+const Channel = require("../models/channel.models");
+const Workspace = require("../models/workspace.models");
+const {
+    response_403,
+    response_500,
+    response_404,
+} = require("../utils/responseCodes.utils");
+
+exports.ensureUserIsMember = async (req, res, next) => {
+    try {
+        const user = req.body.user;
+        const workspace = await Workspace.findById(req.params.id);
+
+        if (!workspace) {
+            return response_404(res, "Workspace doesn't exist");
+        }
+
+        const isMember = await Channel.exists({
```

Comment:
You can get the channel directly from workspace. That will be less time complex i think.

Reply:
Ok Got it, I will do it

Reply:
@automatefreely Can you check now?